src/libraries/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.Windows.cs

```diff
@@ -40,7 +38,7 @@ internal ThreadPoolBoundHandle GetOrAllocateThreadPoolBoundHandle(bool trySkipCo
                 return _iocpBoundHandle;
             }
 
-            lock (_iocpBindingLock)
+            lock (this)
```

Comment:
I thought locking around `this` was frowned upon since users could also lock on it and cause issues. This type is public and doesn't seem to have any mitigation for that possibility. Is the previously mentioned reasoning flawed or doesn't apply in this scenario?

Reply:
Oops. I need to read better.